src/engines/engine-provider.js

```diff
@@ -0,0 +1,83 @@
+//@flow
+import getLogger from '../utils/logger'
+
+/**
+ * Engine Provider
+ * @classdesc
+ */
+class EngineProvider {
+  /**
+   * The logger of the Engine provider.
+   * @member {any} _logger
+   * @static
+   * @private
+   */
+  static _logger: any = getLogger('EngineProvider');
+
+  /**
+   * The Engine adapter registry.
+   * @member {Array<IEngine>} _engineProviders
+   * @static
+   * @private
+   */
+  static _engineProviders: Array<typeof IEngine> = [];
+
+  /**
+   * Add an engine to the registry.
+   * @function register
+   * @param {IEngine} engine -  The engine to register.
+   * @static
+   * @returns {void}
+   */
+  static register(engine: typeof IEngine): void {
+    if (engine) {
+      if (!EngineProvider._engineProviders.includes(engine)) {
+        EngineProvider._logger.debug(`Engine <${engine.id}> has been registered successfully`);
+        EngineProvider._engineProviders.push(engine);
+      } else {
+        EngineProvider._logger.debug(`Engine <${engine.id}> is already registered, do not register again`);
+      }
+    }
+  }
+
+  /**
+   * Remove an engine from the registry.
+   * @function unRegister
+   * @param {IEngine} engine - The engine to unRegister.
+   * @static
+   * @returns {void}
+   */
+  static unRegister(engine: typeof IEngine): void {
+    let index = EngineProvider._engineProviders.indexOf(engine);
+    if (index > -1) {
+      EngineProvider._logger.debug(`Unregistered <${engine.id}> Engine`);
+      EngineProvider._engineProviders.splice(index, 1);
+    }
+  }
+
+  /**
+   * Get the appropriate Engines.
+   * @function getEngines
+   * @returns {Array<IEngine>|null} - The Array of engines, or null if such doesn't exists.
+   * @static
+   */
+  static getEngines(): ?Array<typeof IEngine> {
+    if (EngineProvider._engineProviders.length > 0) {
```

Comment:
why do we need this condition at all? the `EngineProvider._engineProviders` is initialized as an empty array.